trunner/test_runner.py

```diff
@@ -20,6 +24,69 @@ def resolve_project_path():
     return project_dir
 
 
+def init_logdir():
+    """Inits log directories if it's needed."""
+    if not trunner.ctx.logdir:
+        return
+
+    logdir = trunner.ctx.logdir
+    internal_logdir_names = ("out", "in", "inout")
+    if not os.path.isdir(logdir):
+        os.mkdir(logdir)
+
+    for dirname in internal_logdir_names:
+        if not os.path.isdir(f"{logdir}/{dirname}"):
+            os.mkdir(f"{logdir}/{dirname}")
+        # clear logfiles from the whole test campaign
+        else:
+            with open(f"{trunner.ctx.logdir}/{dirname}/test_campaign.log", "w"):
+                break
+
+
+def set_logfiles(dut: Dut):
+    """Sets dut logfiles associated with the pexpect process if needed."""
+
+    if not trunner.ctx.logdir and not is_github_actions():
+        return
+
+    logfile_r, logfile_w, logfile_a = StringIO(""), StringIO(""), StringIO("")
+    dut.set_logfiles(logfile_r, logfile_w, logfile_a)
+
+
+def format_logs_for_gh(logs_in):
+    """Prepares drop-down list for github actions workflows based on provided raw logs"""
+
+    logs = logs_in
```

Comment:
I suggest reaming `logs_in` to just `logs`, since we are doing this anyway in the first statement.